packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/account-page-list.tsx

```diff
@@ -0,0 +1,284 @@
+    "hash-expanded-sidebar-pages",
+    [],
+  );
+
+  const [pagesTreeList, setPagesTreeList] = useState<TreeElement[]>([]);
+  const [pagesFlatList, setPagesFlatList] = useState<TreeElement[]>([]);
+  const [pagesFlatIdList, setPagesFlatIdList] = useState<string[]>([]);
+
+  const [activeId, setActiveId] = useState<UniqueIdentifier | null>(null);
+  const [overId, setOverId] = useState<UniqueIdentifier | null>(null);
+  const [offsetLeft, setOffsetLeft] = useState(0);
+
+  const loading =
+    createUntitledPageLoading ||
+    createSubpageLoading ||
+    reorderLoading ||
+    archivePageLoading;
+
+  // @todo handle loading/error states properly
+  const addPage = useCallback(async () => {
+    if (loading) {
+      return;
+    }
+
+    try {
+      await createUntitledPage();
+    } catch (err) {
+      // eslint-disable-next-line no-console -- TODO: consider using logger
+      console.error("Could not create page: ", err);
+    }
+  }, [createUntitledPage, loading]);
+
+  const handleToggle = (nodeId: string) => {
+    setExpandedPageIds((expandedIds) =>
+      expandedIds.includes(nodeId)
+        ? expandedIds.filter((id) => id !== nodeId)
+        : [...expandedIds, nodeId],
+    );
+  };
+
+  const setSortableList = (
+    list: AccountPage[],
+    expandedPageIdList: string[],
+  ) => {
+    const { treeList, flatList } = getPageList(list, expandedPageIdList);
+    setPagesTreeList(treeList);
+    setPagesFlatList(flatList);
+    setPagesFlatIdList(flatList.map((page) => page.entityId));
+  };
+
+  useMemo(() => {
+    if (!loading) {
+      setSortableList(data, expandedPageIds);
+    }
+  }, [data, expandedPageIds, loading]);
+
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 5,
+      },
+    }),
+  );
+
+  const projected =
+    activeId && overId
+      ? getProjection(pagesFlatList, activeId, overId, offsetLeft, 16)
+      : null;
+
+  const resetState = () => {
+    setOverId(null);
+    setActiveId(null);
+    setOffsetLeft(0);
+    document.body.style.setProperty("cursor", "");
+  };
+
+  const handleDragStart = ({
+    active: { id: activeItemId },
+  }: DragStartEvent) => {
+    if (expandedPageIds.findIndex((id) => id === activeItemId) > -1) {
+      setSortableList(
+        data,
+        expandedPageIds.filter((id) => id !== activeItemId),
+      );
+    }
+    setActiveId(activeItemId);
+    setOverId(activeItemId);
+
+    document.body.style.setProperty("cursor", "grabbing");
+  };
+
+  const handleDragMove = ({ delta }: DragMoveEvent) => {
+    setOffsetLeft(delta.x);
+  };
+
+  const handleDragOver = ({ over }: DragOverEvent) => {
+    setOverId(over?.id ?? null);
+  };
+
+  const handleDragEnd = async ({ active, over }: DragEndEvent) => {
+    resetState();
+
+    if (projected && over) {
+      const { depth, parentPageEntityId } = projected;
+
+      const clonedItems = [...pagesFlatList] as TreeElement[];
```

Comment:
Residual code from previous iterations where I was cloning an AccountPage array instead of a TreeElement array. Removed it in a84edd29298fe75136e917d4d1b5788b504c1c2e